src/common/getters.ts

```diff
@@ -171,16 +190,16 @@ export function getIssuanceDate(
  * @example
  *
  * ```
- * const date = getExpirationDate(gConsentGrant);
+ * const date = getExpirationDate(accessGrant);
  * ```
  *
  * @param vc The Access Grant/Request
- * @returns The issuance date
+ * @returns The expiration date
  */
 export function getExpirationDate(
-  vc: AccessGrantGConsent | AccessRequestGConsent
+  vc: AccessGrantGConsent | AccessRequestGConsent | AccessGrantOdrl
 ): Date | undefined {
-  return vc.expirationDate ? new Date(vc.expirationDate) : undefined;
+  return vc.expirationDate ? new Date(vc.expirationDate as string) : undefined;
```

Comment:
```suggestion
  return typeof vc.expirationDate === "string" ? new Date(vc.expirationDate) : undefined;
```

By explicitly checking the type, you avoid the need for the `as X`

Reply:
What should this function do if it fails to parse the string as a Date?

Reply:
I'd expect it to throw, that's what the call to `new Date(vc.expirationDate)` should do right?

Reply:
I'd expect it to return undefined, why would it throw? An invalid date is as good as no date, surely?

Reply:
Right but you can't mutate what is passed in, this is an issue with the VC provider, they should've validated & formatted the date according to the JSON spec.

Reply:
That's true, this is coming from the server and it is immutable, so throwing doesn't really help actually. I never feel too good about swallowing errors, if something is wrong with the VC it would seem right to bubble it up in some way, and have the consuming app be able to notify the user that something is wrong and they may want to do something about it. From a usefulness perspective, an invalid date is as good as no date, but I guess having a different behaviour allows the app to act accordingly?